src/codegen/doc/mod.rs

```diff
@@ -870,6 +870,32 @@ where
                 )?;
             }
         }
+        // document function's Error
+        let gerror_out_parameter = no_array_length_params
+            .iter()
+            .filter(|param| {
+                param.direction == ParameterDirection::Out
+                    && !doc_ignored_parameters.contains(&param.name)
+                    && param.c_type == "GError**"
+            })
+            .next();
+            
+        if fn_.ret().doc.is_some() || !gerror_out_parameter.name.is_empty() {
```

Comment:
that won't work, the `gerror_out_parameter` is an `Option<T>` which means you can't access `T` until you are sure it is `Some(T)` and not `None`. You want something like 
```suggestion
        if fn_.ret().doc.is_some() || gerror_out_parameter.is_some() {
```
instead

Reply:
🙈